packages/mui-material/src/Snackbar/Snackbar.js

```diff
@@ -182,17 +182,18 @@ const Snackbar = React.forwardRef(function Snackbar(inProps, ref) {
     ownerState,
   });
 
-  const [ClickAwaySlot, clickAwayListenerProps] = useSlot('clickAwayListener', {
-    elementType: ClickAwayListener,
-    externalForwardedProps,
-    getSlotProps: (handlers) => ({
-      onClickAway: (...params) => {
-        handlers.onClickAway?.(...params);
-        onClickAway(...params);
-      },
-    }),
-    ownerState,
-  });
+  const [ClickAwaySlot, { ownerState: clickAwayOwnerStateProp, ...clickAwayListenerProps }] =
```

Comment:
`ownerState` is not a `ClickAwayListener` prop, so proptypes check failed on React 18.